### PR TITLE
restructure docs

### PR DESCRIPTION
**Pull request by deitch**
I pulled a lot of information out of README.md to simplify it and make it easier to consume, moving much into subsidiary documents.

More importantly, I clarified (with much help from @giggsoff ) how various steps work and what they depend on.

In addition, I removed a lot of duplicate information.

The hope is that this makes it much easier to understand and consume for people using it.

I particularly would like an additional review from @kgiori . 

This does _not_ have to be perfect to go in. It should pass the following:

* is not any more incorrect than what was before (e.g. if some of it was out of date)
* is a step forward

I have a major follow-up to start working on docs beginning in `tests/README.md`. That really was the goal of this, to get there. From there, I will document eden as a test and infrastructure harness vs eden as a test API vs eden as tests vs eden as utilities for using inside tests.

### CODE REVIEW

File: docs/eve-images.md [hunk 1]
```diff
@@ -9,38 +9,152 @@ to use a different version, you may want to use a different disk image entirely.
...
+certificates.
+
+### External and Standard
+
+This use case is almost identical to [QEMU and Standard][QEMU and Standard]. The
+only differences are:
+
+* we extract the image after `eden setup` to flash it to our device
+* we start our external device on our own
+* we instruct `eden start` to start everything _except_ EVE via qemu
```
Comment (deitch) [inline docs/eve-images.md:102]: @giggsoff how? What is it that tells `eden start` _not_ to start qemu?

Comment (giggsoff) [reply docs/eve-images.md:102]: Seems, we have `eve.remote` in our config file, which indicates, that `eden start` should ends before starting of qemu/VB/Parallels. It is set to true with GCP or general models.

Comment (deitch) [reply docs/eve-images.md:102]: So `eve.remote=true` means, "the eve is remote and not under eden control, so don't try to start or stop it"?

If so, what is the right invocation to enable it? I _think_ it happens in any of the following cases:

* `eden config add <name> --devmodel general` (automatically sets it)
* `eden config add <name> --devmodel gcp` (automatically sets it)
* `eden config set <name> --key=eve.remote --value=true` (explicitly set it)

Is that correct? Confirm (or correct me), and I will update the PR appropriately.

Comment (giggsoff) [reply docs/eve-images.md:102]: One correction: `GCP`, not `gcp`.  The rest is true

Comment (deitch) [reply docs/eve-images.md:102]: When I set `--devmodel general`, is that just shorthand for `eve config set <name> --key <something> --value general`? What is that setting?

Comment (deitch) [reply docs/eve-images.md:102]: Or does the actual command `--devmodel` set multiple fields, including both `eve.devmodel` and `eve.remote`?

Comment (deitch) [reply docs/eve-images.md]: I even added a section. Take another look @giggsoff

Comment (giggsoff) [reply docs/eve-images.md]: Yes, `--devmodel general` sets multiple options (https://github.com/lf-edge/eden/blob/master/pkg/models/general.go#L25-L33). Will take a look, thank you!

File: README.md [hunk 2]
```diff
@@ -3,657 +3,288 @@
...
+configuration from the controller, and has no console commands that can be used standalone,
+like general-purpose Linux distributions. You use the controller to tell EVE which workloads
+you want to run. EVE, in turn, runs those workloads in containers or VMs.
 
-This machine should be reachable on the network from the edge device and expose port 3333. This machine will host Adam - the controller, Eden - the control layer with CLI, Redis - the log database, Eserver  - the image/file database for Eve to download from. The alternative is to use some commercial controller.
+Eve supports the following workload formats:
 
-Note that EVE project by itself without a controller is useless in practice. Eve by itself doesn't have any good options/commands to be used as some other OS like Ubuntu. You don't run workloads on Eve kernel. Eve core is separated from apps - the apps are either containers or VMs.
+* OCI image from any OCI compliant registry
+* OS image in qcow2 format
```
Comment (giggsoff) [inline README.md:26]: Seems, it is different formats. OCI image is only about rootfs update (raw is also about rootfs, but also about disk). QCOW2 (also raw, VDI, Parallels) are formats of disk with EVE installed.

File: README.md [hunk 3]
```diff
@@ -3,657 +3,288 @@
...
-make build-tests
-./eden config add default
-./eden setup
-source ~/.eden/activate.sh
-./eden start
-./eden eve onboard
-./eden pod deploy docker://nginx -p 8028:80 --mount=src=./data/helloeve,dst=/usr/share/nginx/html
-./eden status
-```
+Alternatively, Eden supports using a commercial controller.
```
Comment (giggsoff) [inline README.md:86]: Not for now. Actually Eden can create an image for onboarding into commercial controller, but all other things are out of scope.

File: docs/applications.md [hunk 4]
```diff
@@ -92,3 +146,131 @@ is the volume from list.
...
+eden deploys its own http server and docker registry, to simplify presenting
+local files and container images to a running EVE device.
+
+Each source has its default format, but you can override it with
+the `--format` flag (`container`,`qcow2` or `raw`). The defaults are:
+
+* `docker://` - OCI image
+* `http://` - VM qcow2
+* `https://` - VM qcow2 image
+* `file://` - VM qcow2 image
```
Comment (giggsoff) [inline docs/applications.md:174]: Can you please add "from local file".

File: docs/eve-images.md [hunk 5]
```diff
@@ -9,38 +9,176 @@ to use a different version, you may want to use a different disk image entirely.
...
-   * device certificate
-1. Run the docker image with the config partition mounted to:
-   1. Generate a disk image - raw for RPi, qcow2 for everything else - overriding
-   the config partition with the mounted config directory
-   1. Extract the disk image from the docker image, e.g. `live.qcow2`
-1. Save the extracted disk image to a local cache, normally
-`$PWD/dist/<name>-images/eve/` (where `<name>` is the current context name).
+1. create a named context to store all of your configuration - `eden config add <name>`
+1. (optional) set options for the context - `eden config set <name> [options...]`
+1. run setup - `eden setup`, which extracts an eve-os qcow2 disk image from the docker image named in the context
```
Comment (giggsoff) [inline docs/eve-images.md:22]: Seems, you also should add `--config <name>` here and below.

**Review by giggsoff — CHANGES_REQUESTED** [review]
Please see my comments, take a look onto yetus errors and pay attention to write `--config <name>` when you write about custom context.

File: README.md (hunk 2 above)
Comment (deitch) [reply README.md:26]: @giggsoff These are the _workload_ formats, not the _boot_ formats. See L23 right before. It is not authoritative, just to give a feel. EVE definitely supports running OCI containers.

File: README.md (hunk 3 above)
Comment (deitch) [reply README.md:86]: Yeah, but that is too complex for the README. I fixed it. Take a look as soon as I push.

File: docs/applications.md (hunk 4 above)
Comment (deitch) [reply docs/applications.md:174]: Sure, I added information to the whole section.

File: docs/eve-images.md (hunk 5 above)
Comment (deitch) [reply docs/eve-images.md:22]: Added

**Comment by deitch** [comment]
Got most of the yetus complaints

**Comment by deitch** [comment]
yetus cannot handle code blocks embedded in lists. Ignore them.

File: README.md (hunk 2 above)
Comment (giggsoff) [reply README.md:26]: got it, thanks for clarification

File: docs/eve-images.md (hunk 5 above)
Comment (giggsoff) [reply docs/eve-images.md]: Thank you!

**Review by giggsoff — APPROVED** [review]
Thank you for organizing it!

**Comment by deitch** [comment]
> Thank you for organizing it!

A pleasure, and thank you for helping me through it.

**Review by rvs — COMMENTED** [review]
LGTM! Late with this feedback -- but thanks a million @deitch for doing this!

**Pull request merged**